internal/cache/file/cache_handler_test.go

```diff
@@ -523,112 +562,117 @@ func (chrT *cacheHandlerTest) Test_GetCacheHandle_ConcurrentDifferentFiles() {
 	objectContent := []byte("content of object_1")
-	minObject := chrT.getMinObject("object_1", objectContent)
-	cacheHandle, err := chrT.cacheHandler.GetCacheHandle(minObject, chrT.bucket, false, 0)
-	require.NoError(chrT.T(), err)
+	minObject := createObject(t, chTestArgs.bucket, "object_1", objectContent)
+	cacheHandle, err := chTestArgs.cacheHandler.GetCacheHandle(minObject, chTestArgs.bucket, false, 0)
+	require.NoError(t, err)
 	buf := make([]byte, 3)
 	ctx := context.Background()
 	// Read to populate cache
-	_, cacheHit, err := cacheHandle.Read(ctx, chrT.bucket, minObject, 0, buf)
-	require.NoError(chrT.T(), err)
-	require.Equal(chrT.T(), false, cacheHit)
-	require.Equal(chrT.T(), string(objectContent[:3]), string(buf))
-	require.Nil(chrT.T(), cacheHandle.Close())
+	_, cacheHit, err := cacheHandle.Read(ctx, chTestArgs.bucket, minObject, 0, buf)
+	require.NoError(t, err)
+	require.Equal(t, string(objectContent[:3]), string(buf))
+	require.Equal(t, false, cacheHit)
+	require.Equal(t, nil, cacheHandle.Close())
 	// Open cache file before invalidation
-	objectPath := util.GetObjectPath(chrT.bucket.Name(), minObject.Name)
-	downloadPath := util.GetDownloadPath(chrT.cacheDir, objectPath)
+	objectPath := util.GetObjectPath(chTestArgs.bucket.Name(), minObject.Name)
+	downloadPath := util.GetDownloadPath(chTestArgs.cacheDir, objectPath)
 	file, err := os.OpenFile(downloadPath, os.O_RDONLY, 0600)
-	require.NoError(chrT.T(), err)
-	_, err = file.Read(buf)
-	require.NoError(chrT.T(), err)
-	require.Equal(chrT.T(), string(objectContent[:3]), string(buf))
+	require.NoError(t, err)
 
-	err = chrT.cacheHandler.InvalidateCache(minObject.Name, chrT.bucket.Name())
+	err = chTestArgs.cacheHandler.InvalidateCache(minObject.Name, chTestArgs.bucket.Name())
 
-	assert.NoError(chrT.T(), err)
+	assert.NoError(t, err)
 	// Reading from the open file handle should fail as the file is truncated.
 	_, err = file.Read(buf)
-	assert.NotNil(chrT.T(), err)
-	assert.Equal(chrT.T(), io.EOF, err)
+	assert.ErrorIs(t, err, io.EOF)
 }
 
-func (chrT *cacheHandlerTest) Test_InvalidateCache_ConcurrentSameFile() {
-	existingJob := chrT.getDownloadJobForTestObject()
-	require.Equal(chrT.T(), downloader.NotStarted, existingJob.GetStatus().Name)
-	require.True(chrT.T(), chrT.isEntryInFileInfoCache(chrT.object.Name, chrT.bucket.Name()))
+func Test_InvalidateCache_ConcurrentSameFile(t *testing.T) {
+	cacheDir := path.Join(os.Getenv("HOME"), "CacheHandlerTest/dir")
+	chTestArgs := initializeCacheHandlerTestArgs(t, &config.FileCacheConfig{EnableCRC: true}, cacheDir)
+	existingJob := getDownloadJobForTestObject(t, chTestArgs)
+	require.Equal(t, downloader.NotStarted, existingJob.GetStatus().Name)
+	require.True(t, isEntryInFileInfoCache(t, chTestArgs.cache, chTestArgs.object.Name, chTestArgs.bucket.Name()))
 	wg := sync.WaitGroup{}
-	InvalidateCacheTestFun := func() {
+	InvalidateCacheTestFun := func(t *testing.T) {
```

Comment:
This being an inner function should already have access to "t" from the outer scope. So, we probably don't need this parameter.